src/datasets/utils/streaming_download_manager.py

```diff
@@ -15,6 +15,7 @@
 
 logger = get_logger(__name__)
 BASE_KNOWN_EXTENSIONS = ["txt", "csv", "json", "jsonl", "tsv", "conll"]
+UNSUPPORTED_ARCHIVE_EXTENSIONS_FOR_STREAMING = ["tar", "xz", "rar", "zst"]
```

Comment:
Here also we should have the list of supported (and unsupported) extension/file-format clearly stated in the doc

Reply:
I'm a bit surprised you do a black-list instead of a white-list here

Reply:
This is because most scripts call `download_and_extract` even for files that don't need extraction. We shouldn't raise an error for an uncompressed file. If we had a white list and you passed un uncompressed file, it wouldn't find any supported compression extension in the filename, so it would fail.

Reply:
in the future we'll fallback on reading the header of the file to know if it's a compressed file or not